ush/gaussian_sfcanl.sh

```diff
@@ -133,42 +133,59 @@ export PGM=$GAUSFCANLEXE
 export pgm=$PGM
 $LOGSCRIPT
 
-iy=${PDY:0:4}
-im=${PDY:4:2}
-id=${PDY:6:2}
-ih=${cyc}
+gcycle_dates=("${PDY}${cyc}")
+if [[ "${DOIAU:-}" == "YES" ]]; then  # Create Gaussian sfcanl file at beginning of window
+  half_window=$(( assim_freq / 2 ))
+  BDATE=$(date --utc -d "${PDY} ${cyc} - ${half_window} hours" +%Y%m%d%H)
+  gcycle_dates+=("${BDATE}")
+fi
+
+for hr in "${!gcycle_dates[@]}"; do
```

Comment:
```suggestion
for ii in "${!gcycle_dates[@]}"; do
```
This is an index, not hour.